### PR TITLE
Release 3.3.2

### PR DESCRIPTION
**Pull request by awholegunch**
- Fixes broken links
- Adds a [release note for https://issues.cask.co/browse/CDAP-5019](http://builds.cask.co/artifact/CDAP-DOB123/shared/build-6/Docs-HTML/3.3.1/en/reference-manual/release-notes.html#release-3-3-2)
- Adds a [release note for https://issues.cask.co/browse/CDAP-5047](http://builds.cask.co/artifact/CDAP-DOB123/shared/build-6/Docs-HTML/3.3.1/en/reference-manual/release-notes.html#release-3-3-2)
- Adds the documentation for [Azure Blob Storage plugin](http://builds.cask.co/artifact/CDAP-DOB123/shared/build-6/Docs-HTML/3.3.1/en/cdap-apps/hydrator/hydrator-plugins/batchsources/azureblobstore.html).

Passes [Quick Build 6](https://builds.cask.co/browse/CDAP-DOB123-5)

Updates the CDH version on:

- [FAQ: Hadoop Versions](http://builds.cask.co/artifact/CDAP-DOB123/shared/build-6/Docs-HTML/3.3.1/en/faqs/general.html#what-hadoop-distributions-can-cdap-run-on)
- [Hadoop Compatibility](http://builds.cask.co/artifact/CDAP-DOB123/shared/build-6/Docs-HTML/3.3.1/en/admin-manual/hadoop-compatibility.html)
- [CM Installation](http://builds.cask.co/artifact/CDAP-DOB123/shared/build-6/Docs-HTML/3.3.1/en/admin-manual/installation/cloudera.html#downloading-and-distributing-packages)

### CODE REVIEW

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 1]
```diff
@@ -23,6 +23,15 @@ Cask Data Application Platform Release Notes
    :backlinks: none
    :depth: 2
 
+`Release 3.3.2 <http://docs.cask.co/cdap/3.3.2/index.html>`__
+=============================================================
+
+Bug Fixes
+---------
+- `CDAP-5019 <https://issues.cask.co/browse/CDAP-5019>`__ - Fixed an issue with
```
Comment (anwar6953) [inline cdap-docs/reference-manual/source/release-notes.rst]: @chtyim If you're satisfied with this release note message, let's also put it on the JIRA and merge this PR.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 2]
```diff
@@ -23,6 +23,15 @@ Cask Data Application Platform Release Notes
...
+  launching Spark jobs on Cloudera Manager clusters when using Spark 1.4 or greater.
```
Comment (chtyim) [inline cdap-docs/reference-manual/source/release-notes.rst]: Do we call it "Cloudera Manager" clusters or "Clusters created with Cloudera Manager"?

Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: The first question: Is this a CDH issue, or a CM issue? Is it any CDH cluster, both managed and not managed with CM, or is it restricted to CDH cluster managed with CM?

We can call it "CDH clusters managed with Cloudera Manager". See http://docs.cask.co/cdap/3.3.1/en/admin-manual/installation/packages.html.

But if it is really CDH clusters, then we should say:
"Cloudera Data Hub clusters when using Spark 1.4 or greater."

Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Reviewing the original issue, looks like we should go with the former.

**Comment by bdmogal** [comment]
@awholegunch please include [CDAP-5047](https://issues.cask.co/browse/CDAP-5047) as an improvement in the release notes

**Comment by anwar6953** [comment]
@chtyim if you're satisfied with the release note message, let's merge this.

**Comment by bdmogal** [comment]
The note about Azure Blob Storage LGTM

**Comment by awholegunch** [comment]
@bdmogal I added an additional "Microsoft" to the release notes...

**Comment by prinam** [comment]
Please add the following JIRAs to the release notes:

https://issues.cask.co/browse/CDAP-5092
https://issues.cask.co/browse/CDAP-5094
https://issues.cask.co/browse/CDAP-5137

**Comment by anwar6953** [comment]
Also this - https://issues.cask.co/browse/CDAP-5134
Sorry, I had the fixVersion wrong on it, so maybe it didn't get captured; fixed it now.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 3]
```diff
@@ -23,6 +23,31 @@ Cask Data Application Platform Release Notes
...
+-------
+- `CDAP-4967 <https://issues.cask.co/browse/CDAP-4967>`__ - Fixed a schema-parsing bug
+  that prevented the use of schemas where a record is used both as a top-level field and
+  also used inside a different record field.
+- `CDAP-5019 <https://issues.cask.co/browse/CDAP-5019>`__ - Worked around two issues
+  (`SPARK-13441 <https://issues.apache.org/jira/browse/SPARK-13441>`__
+  and `YARN-4727 <https://issues.apache.org/jira/browse/YARN-4727>`__) that prevented
+  launching Spark jobs on Cloudera Data Hub clusters managed with Cloudera Manager 
+  when using Spark 1.4 or greater.
+- `CDAP-5063 <https://issues.cask.co/browse/CDAP-5063>`__ - Fixed a problem with
```
Comment (sreevatsanraman) [inline cdap-docs/reference-manual/source/release-notes.rst]: This is specific to Ambari clusters. We should highlight that.

**Comment by sreevatsanraman** [comment]
one comment. rest LGTM. Fix and merge.

**Comment by anwar6953** [comment]
This JIRA that @prinam mentioned is not yet included in this PR:
https://issues.cask.co/browse/CDAP-5092

@chtyim Can you help construct the release note for the fix that is going into 3.3.2?

**Comment by anwar6953** [comment]
LGTM

**Comment by awholegunch** [comment]
LGTM2

**Comment by anwar6953** [comment]
Squashed commits; merging.

**Pull request merged**